VersionControlUI/src/DiffManager.cpp

```diff
@@ -411,34 +411,63 @@ void DiffManager::showDiff(QString oldVersion, QString newVersion)
 	diffViewItem->setZoomLabelsEnabled(false);
 
 	int row = 0;
-	auto diffFrames = createDiffFrames(diffSetup, changedNodesToVisualize);
+	auto diffFrames = createDiffFrames(diffSetup, changedNodesToVisualize, changesWithNodes);
 	for (auto diffFrame : diffFrames)
 		diffViewItem->insertNode(diffFrame, {row++, 0});
 
+	createOverlaysForChanges(changesWithNodes, diffViewItem, diffSetup, diffFrames.first());
+
+	// switch to the newly created view
+	Visualization::VisualizationManager::instance().mainScene()->viewItems()->switchToView(diffViewItem);
+}
+
+void DiffManager::createOverlaysForChanges(QList<ChangeWithNodes> changesWithNodes, Visualization::ViewItem* viewItem,
+														 DiffSetup diffSetup, VersionControlUI::DiffFrame* diffFrame)
+{
 	auto nameChangeInfo = computeNameChangeInformation(diffSetup);
 
-	// create visualization for changes
 	Visualization::VisualizationManager::instance().mainScene()->addPostEventAction(
-								  [diffViewItem, changesWithNodes, diffSetup, nameChangeInfo, diffFrames]() {
-		createOverlaysForChanges(diffViewItem, changesWithNodes);
+								  [viewItem, changesWithNodes, diffSetup, nameChangeInfo, diffFrame]() {
+		createOverlaysForChanges(viewItem, changesWithNodes);
 		auto message = createHTMLCommitInfo(diffSetup.repository_, diffSetup.newVersion_);
 		if (!nameChangeInfo.isEmpty()) message += "<br/><br/>" + nameChangeInfo;
-		auto overlay = new Visualization::MessageOverlay{diffViewItem,
-				[diffViewItem, message, diffFrames](Visualization::MessageOverlay* overlay)
+		auto overlay = new Visualization::MessageOverlay{viewItem,
+				[viewItem, message, diffFrame](Visualization::MessageOverlay* overlay)
 		{
-			auto vDiffFrame = DCast<VersionControlUI::VDiffFrame>(diffViewItem->findVisualizationOf
-																					(diffFrames.first()));
+			auto vDiffFrame = DCast<VersionControlUI::VDiffFrame>(viewItem->findVisualizationOf
+																					(diffFrame));
 			overlay->setPos(vDiffFrame->scenePos().x(),
 								 vDiffFrame->scenePos().y()-overlay->heightInScene());
 			return message;
 		}, Visualization::MessageOverlay::itemStyles().get("info"), true};
 
 
-		diffViewItem->addOverlay(overlay, "DiffInfoMessageOverlay");
+		viewItem->addOverlay(overlay, "DiffInfoMessageOverlay");
 	});
+}
 
-	// switch to the newly created view
-	Visualization::VisualizationManager::instance().mainScene()->viewItems()->switchToView(diffViewItem);
+QList<DiffFrame*> DiffManager::computeDiffFramesWithOverlays(QString oldVersion, QString newVersion,
```

Comment:
Somehow the name of this method does not reflect what's going on. The body seems to do a lot more that just "compute", for example it creates new version managers and registers them on the scene.

Also do you mean `with` or `and` in the method name? `With` implies that this method computes only those DiffFrames that have an Overlay. `And` implies that it's doing both. It seems like the second is the case here.

There are quite a few methods with similar names and we need to make it clearer what method does what:
`computeDiffFramesWithOverlays`, `computeDiff`, createDiffFrames`.